src/amp-story-player/amp-story-player-impl.js

```diff
@@ -730,20 +771,79 @@ export class AmpStoryPlayer {
 
   /**
    * Indicates the player changed story.
+   * @param {!Object} data
    * @private
    */
-  signalNavigation_() {
+  signalNavigation_(data) {
+    const event = createCustomEvent(this.win_, 'navigation', data);
+    this.element_.dispatchEvent(event);
+  }
+
+  /**
+   * Triggers when swithing from one story to another.
+   * @private
+   */
+  onNavigation_() {
     const index = this.currentIdx_;
     const remaining = this.stories_.length - this.currentIdx_ - 1;
-    const event = createCustomEvent(
-      this.win_,
-      'navigation',
-      dict({
-        'index': index,
-        'remaining': remaining,
-      })
-    );
-    this.element_.dispatchEvent(event);
+    const navigation = {
+      'index': index,
+      'remaining': remaining,
+    };
+
+    this.signalNavigation_(navigation);
+    this.maybeFetchMoreStories_(navigation);
+  }
+
+  /**
+   * Fetches more stories if appropiate.
+   * @param {!Object} data
+   */
+  maybeFetchMoreStories_(data) {
+    if (
+      data.remaining <= FETCH_STORIES_THRESHOLD &&
+      this.isFetchingEnabled_()
+    ) {
+      this.fetchStories_()
+        .then((stories) => {
+          this.add(stories);
+        })
+        .catch((reason) => {
+          console /*OK*/
+            .error(`[${TAG}] `, reason);
+        });
+    }
+  }
+
+  /**
+   * Checks if fetching more stories is enabled and validates the configuration.
+   * @return {boolean}
+   * @private
+   */
+  isFetchingEnabled_() {
```

Comment:
Nit: please be more accurate in the method name, we could be fetching more stuff in the future (poster images for following stories, configuration, ...)